File: gamedex-core/src/main/java/com/github/ykrasik/gamedex/core/config/type/GameSort.java
package com.github.ykrasik.gamedex.core.config.type;

/**
 * @author Yevgeny Krasik
 */
public enum GameSort {
    NAME_ASC("Name \u2191"),
    NAME_DESC("Name \u2193"),
    CRITIC_SCORE_ASC("Critic Score \u2191"),
    CRITIC_SCORE_DESC("Critic Score \u2193"),
    USER_SCORE_ASC("User Score \u2191"),
    USER_SCORE_DESC("User Score \u2193"),
    RELEASE_DATE_ASC("Release Date \u2191"),
    RELEASE_DATE_DESC("Release Date \u2193"),
    DATE_ADDED_ASC("Date Added \u2191"),
    DATE_ADDED_DESC("Date Added \u2193");

    private final String displayName;

    GameSort(String displayName) {
        this.displayName = displayName;
    }

    public String getDisplayName() {
        return displayName;
    }

    @Override
    public String toString() {
        return displayName;
    }
}

File: gamedex-app/src/main/kotlin/com/gitlab/ykrasik/gamedex/preferences/GameSort.kt
package com.gitlab.ykrasik.gamedex.preferences

/**
 * User: ykrasik
 * Date: 23/04/2017
 * Time: 12:49
 */
// TODO: I don't like displayName on the enum.
enum class GameSort constructor(val displayName: String) {
    nameAsc("Name \u2191"),
    nameDesc("Name \u2193"),
    criticScoreAsc("Critic Score \u2191"),
    criticScoreDesc("Critic Score \u2193"),
    userScoreAsc("User Score \u2191"),
    userScoreDesc("User Score \u2193"),
    minScoreAsc("Min Score \u2191"),
    minScoreDesc("Min Score \u2193"),
    avgScoreAsc("Average Score \u2191"),
    avgScoreDesc("Average Score \u2193"),
    releaseDateAsc("Release Date \u2191"),
    releaseDateDesc("Release Date \u2193"),
    dateAddedAsc("Date Added \u2191"),
    dateAddedDesc("Date Added \u2193");

    override fun toString() = displayName
    
    private companion object {
        final val downArrow = "\u2191"
    }
}